icu4c/source/i18n/tibetancal.cpp

```diff
@@ -0,0 +1,538 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+/*
+ * Copyright (C) 2003-2021, International Business Machines Corporation
+ * and others. All Rights Reserved.
+ ******************************************************************************
+ *
+ * File TIBETANCAL.CPP
+ *****************************************************************************
+ */
+
+#include "tibetancal.h"
+#include <stdlib.h>
+#if !UCONFIG_NO_FORMATTING
+
+#include "umutex.h"
+#include <float.h>
+#include "gregoimp.h" // Math
+#include "uhash.h"
+
+// Debugging
+#ifdef U_DEBUG_TIBETANCAL
+#include <stdio.h>
+#include <stdarg.h>
+
+#endif
+
+U_NAMESPACE_BEGIN
+
+// Implementation of the TibetanCalendar class
+
```

Comment:
Please add the following in the comments as references:
// Ref: [Janson] Janson, Svante. "Tibetan calendar mathematics." arXiv preprint arXiv:1401.6285 (2014).
// http://www2.math.uu.se/~svante/papers/calendars/tibet.pdf

Reply:
Well, this reference could be added but the basis of the code is really 

`[Henning] Henning, Edward. "Kālacakra and the Tibetan Calendar." American Institute of Buddhist Studies, New York, 2007.`

but there should be no difference